ResearchKit/Common/ORKPieChartView.m

```diff
@@ -0,0 +1,478 @@
+        CGFloat value = ((NSNumber *)self.normalizedValues[idx]).floatValue;
+        
+        if (value != 0) {
+            
+            if (idx == 0) {
+                segmentLayer.strokeStart = 0.0;
+            } else {
+                segmentLayer.strokeStart = cumulativeValue;
+            }
+            
+            segmentLayer.strokeEnd = cumulativeValue;
+            [self.circleLayer addSublayer:segmentLayer];
+            
+            if (self.shouldAnimate) {
+                
+                CABasicAnimation *strokeAnimation = [CABasicAnimation animationWithKeyPath:@"strokeEnd"];
+                strokeAnimation.fromValue = @(segmentLayer.strokeStart);
+                strokeAnimation.toValue = @(cumulativeValue + value);
+                strokeAnimation.duration = _animationDuration + 0.1;
+                strokeAnimation.removedOnCompletion = NO;
+                strokeAnimation.fillMode = kCAFillModeForwards;
+                strokeAnimation.timingFunction = [CAMediaTimingFunction functionWithName:kCAMediaTimingFunctionEaseInEaseOut];
+                [segmentLayer addAnimation:strokeAnimation forKey:@"strokeAnimation"];
+            }
+            else {
+                segmentLayer.strokeEnd = cumulativeValue + value;
+            }
+        }
+        cumulativeValue += value;
+    }
+}
+
+- (void)drawPercentageLabels {
+    CGFloat cumulativeValue = 0;
+    NSMutableArray * textLayers = [NSMutableArray array];
+    
+    for (NSInteger idx = 0; idx < [self numberOfSegments]; idx++) {
+        CGFloat value = ((NSNumber *)self.normalizedValues[idx]).floatValue;
+        
+        if (value != 0) {
+            CGFloat angle = (value/2 + cumulativeValue) * M_PI * 2;
+            
+            if (!self.shouldDrawClockwise) {
+                angle = (value/2 + cumulativeValue) * - M_PI * 2;
+            }
+            
+            CGPoint labelCenter = [self getCirclePointForAngle: angle];
+            
+            NSString *text = [NSString stringWithFormat:@"%0.0f%%", (value < .01) ? 1 :value * 100];
+            CATextLayer *textLayer = [CATextLayer layer];
+            textLayer.string = text;
+            textLayer.fontSize = 14.0;
+            textLayer.foregroundColor = [self colorForSegmentAtIndex:idx].CGColor;
+            
+            CGFloat textWidth = [text boundingRectWithSize:CGSizeMake(100, 21) options:NSStringDrawingUsesFontLeading|NSStringDrawingUsesLineFragmentOrigin attributes:@{NSFontAttributeName:[UIFont fontWithName:@"HelveticaNeue" size:textLayer.fontSize]} context:nil].size.width;
+            
+            textLayer.frame = CGRectMake(0, 0, textWidth, 21);
+            textLayer.position = labelCenter;
+            textLayer.alignmentMode = @"center";
+            textLayer.contentsScale = [[UIScreen mainScreen] scale];
+            
+            NSMutableDictionary * layerData = [@{@"layer" : textLayer, @"angle": [NSNumber numberWithFloat:angle]} mutableCopy];
```

Comment:
Always have one space between variable type and `*`, and no space between `*` and variable name. There are a few wrong ones around.